### PR TITLE
Create SSH service

### PR DESCRIPTION
**Pull request by rcobbins**

### CODE REVIEW

File: internal/httperror/httperror.go [hunk 1]
```diff
@@ -602,6 +659,8 @@ func DecodeSigningKeyError(err string) HTTPError {
 	switch getCallerFunctionName() {
 	case "SetIntermediateCertificate":
 		errorCode = "CPKISI009"
+	case "CreateSSHCertificate":
+		errorCode = "CSSHCC012"
```
Comment (AndrewCopeland) [inline internal/httperror/httperror.go:663]: Should this error code be the same as the one below?

File: pkg/pkiaas/api/ssh_controller.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+	err = storage.GetAccessControl().CreateSSHCertificate(authHeader, certReq.TemplateName)
+	if err != nil {
+		httpErr := httperror.InvalidAuthz(err.Error())
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+
+	sshCert, httpErr := ssh.CreateSSHCertificate(certReq, storage)
+	if httpErr != (httperror.HTTPError{}) {
```
Comment (AndrewCopeland) [inline pkg/pkiaas/api/ssh_controller.go:273]: This doesn't seem right. Shouldn't we having something like:
```
		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
                return
```

File: internal/backend/conjur/ssh_templates.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package conjur
```
Comment (AndrewCopeland) [inline internal/backend/conjur/ssh_templates.go:1]: So we will not be writing the SSH certificates to the backend?

File: internal/ssh/helpers.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
+package ssh
+
+import (
+	"errors"
+	"net"
+	"regexp"
+
+	"github.com/infamousjoeg/cyberark-aam-pkiaas/internal/types"
+)
+
```
Comment (AndrewCopeland) [inline internal/ssh/helpers.go:10]: Could we test the Validate functions?

File: vendor/modules.txt [hunk 5]
```diff
@@ -6,16 +6,34 @@ github.com/cyberark/conjur-api-go/conjurapi/authn
 github.com/cyberark/conjur-api-go/conjurapi/logging
 github.com/cyberark/conjur-api-go/conjurapi/response
 # github.com/golang/gddo v0.0.0-20200715224205-051695c33a3f
+<<<<<<< HEAD
+## explicit
```
Comment (AndrewCopeland) [inline vendor/modules.txt:10]: This needs to be resolved

File: internal/backend/conjur/ssh_templates.go (hunk 3 above)
Comment (rcobbins) [reply internal/backend/conjur/ssh_templates.go:1]: That's correct, there is no centralized system for SSH key management. There is a concept called KRL that is similar to the CRL, but it is defined on each local system for the SSH server to manage.

File: internal/httperror/httperror.go (hunk 1 above)
Comment (rcobbins) [reply internal/httperror/httperror.go:663]: No, I'll figure out the appropriate numbers and fix them... That was an oversight

File: internal/ssh/helpers.go (hunk 4 above)
Comment (rcobbins) [reply internal/ssh/helpers.go:10]: I'm going to start the unit tests for the SSH capabilities later this week, but yes :)

File: pkg/pkiaas/api/ssh_controller.go (hunk 2 above)
Comment (rcobbins) [reply pkg/pkiaas/api/ssh_controller.go:273]: Yep, this is why you're the best 🥇

**Review by AndrewCopeland — APPROVED** [review]

**Pull request merged**